112-Strimzi-v1-CRD-API-and-1.0.0-release.md

```diff
@@ -0,0 +1,441 @@
+
+### Upgrade path
+
+Users will have the time between the 0.49 and 0.52/1.0.0 releases to migrate their custom resources to the `v1` API.
+The only slight exception will be the `KafkaUser` resources.
+While users will be able to keep using the older versions of the `KafkaUser` API between 0.49.0 and 0.52.0/1.0.0, they will not be able to use the deprecated field `operation` in the ACL rule definition.
+Any resources using this field will be rejected by the User Operator as invalid (however, the User itself will not be deleted from the Apache Kafka cluster).
+This is because the User Operator will be already using the new `v1` API internally.
+While technically the same applies to the Topic Operator and the `KafkaTopic` resources, the `KafkaTopic` resource has no deprecated fields that are being removed in the `v1` API version.
+So there is no similar impact on the `KafkaTopic` resources as on the `KafkaUser` resources.
+
+The upgrade process does not require including Strimzi 0.49.0 or 0.52.0/1.0.0 specifically.
+However, upgrade paths must include:
+* **At least one of Strimzi 0.49.0, 0.50.0, or 0.51.0**, which introduces the `v1` API and transitions the User and Topic Operators to use it.
+* **Strimzi 0.52.0/1.0.0 or newer**, which removes the old API versions and transitions the Cluster Operator to the `v1` API.
+
+For example, Strimzi users can:
+* Start with Strimzi 0.47.0
+* Upgrade to Strimzi 0.50.0
+    * Before this upgrade, they should make sure they do not use the `operation` field in the `KafkaUser` resource
+* Upgrade to Strimzi 1.1.0
+    * CRD conversion of all custom resources to the `v1` API will need to be done for all custom resources before this upgrade
+
+### First step
+
+In the first phase, the Strimzi CRDs will be modified to add the new `v1` API to the existing CRDs.
+The details of how the `v1` API will look for the different resources and how it will differ from the existing API versions are described in a separate section below.
+In the first phase, we will also deprecate the old CRD API versions (set the flag `deprecated: true`).
+The CRD API version deprecation is a flag in the CRD resource.
+Deprecated versions can still be used, but Kubernetes will issue a warning when the deprecated API is used.
+
+Most of the CRDs will continue to use the `v1beta2` version as the stored version in the first phase.
+Only the `KafkaUser` and `KafkaTopic` CRDs will already switch to `v1` as the stored version.
+With the exception of the `KafkaUser` resource, users are not expected to do anything in the first phase other than upgrade the CRDs and the other Strimzi resources.
+
+The `Kafka` CRD YAML with both `v1beta2` and `v1` versions is too big and it would not be possible to use it with `kubectl apply` because of its size.
+Instead of updating all our docs and forcing users to use `kubectl creat` / `kubectl replace` instead of `kubectl apply`, we will generate the `Kafka` CRD with the `description` fields.
```

Comment:
I am confused now, because you commented:

> But basically, if the CRD is too big, kubectl apply will nto work for it. So we make it smaller by leaving out the API descriptions while having both v1beta2 and v1 versions.

So AFAIU "without the description fields" while here I read "with the description fields".
Also, isn't it good having descriptions in the CRD? Isn't it something you can get via the `kubectl describe` instead of going to the doc?

Reply:
That is a typo, yes. It should be `without`.

Reply:
While removing descriptions helps now, it could just delay the problem later if the CRDs increase later. Anyway I have not strong opinion on it. Curious to know what other @strimzi/maintainers think about it.